libraries/Adapters/Microsoft.Bot.Builder.Adapters.Slack/SlackHelper.cs

```diff
@@ -0,0 +1,356 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Net;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using SlackAPI;
+
+#if SIGNASSEMBLY
+[assembly: InternalsVisibleTo("Microsoft.Bot.Builder.Adapters.Twilio.Tests, PublicKey=0024000004800000940000000602000000240000525341310004000001000100b5fc90e7027f67871e773a8fde8938c81dd402ba65b9201d60593e96c492651e889cc13f1415ebb53fac1131ae0bd333c5ee6021672d9718ea31a8aebd0da0072f25d87dba6fc90ffd598ed4da35e44c398c454307e8e33b8426143daec9f596836f97c8f74750e5975c64e2189f45def46b2a2b1247adc3652bf5c308055da9")]
+#else
+[assembly: InternalsVisibleTo("Microsoft.Bot.Builder.Adapters.Slack.Tests")]
+#endif
+
+namespace Microsoft.Bot.Builder.Adapters.Slack
+{
+    internal static class SlackHelper
+    {
+        /// <summary>
+        /// Formats a BotBuilder activity into an outgoing Slack message.
+        /// </summary>
+        /// <param name="activity">A BotBuilder Activity object.</param>
+        /// <returns>A Slack message object with {text, attachments, channel, thread ts} as well as any fields found in activity.channelData.</returns>
+        public static NewSlackMessage ActivityToSlack(Activity activity)
+        {
+            if (activity == null)
+            {
+                return null;
```

Comment:
Presumably this should never be null - why not throw an ArgumentNullException?

Reply:
Regarding the _ArgumentNullException_ for the helper method's parameters, we followed Chris's [recommendation](https://github.com/microsoft/botbuilder-dotnet/pull/1262#discussion_r246097901) to not to force the developers to handle exceptions.
But, if you consider that the exception is the right approach here, we'll change it.

Reply:
Spoken we Chris / Gabo. This should be throwing the ArgumentNullException and not returning null. Please amend. Thanks.

Reply:
Done. Thanks!